discovery-provider/src/tasks/index.py

```diff
@@ -330,45 +426,62 @@ def fetch_ipfs_metadata(
         .all()
     )
 
-    ipfs_metadata = {}
-    with concurrent.futures.ThreadPoolExecutor(max_workers=8) as executor:
-        futures = []
-        futures_map = {}
-        for cid, txhash in cids:
-            metadata_format = (
-                track_metadata_format
-                if cid_type[cid] == "track"
-                else user_metadata_format
+    cid_metadata = {}  # cid -> metadata
+
+    logger.info(f"isaac CIDs {cids_txhash_set}")
+    # first attempt - fetch all CIDs from replica set
+    asyncio.run(
+        fetch_metadata_from_gateway_endpoints(
+            cid_metadata,
+            cids_txhash_set,
+            cid_to_user_id,
+            user_to_replica_set,
+            None,
+            cid_type,
+            block_hash,
+            block_number,
+            fetch_from_replica_set=True,
+        )
+    )
+    logger.info(f"isaac fetched from repica set {cid_metadata}")
+    # second attempt - fetch missing CIDs from other cnodes
+    if len(cid_metadata) != len(cids_txhash_set):
+        eth_web3 = update_task.eth_web3
+        shared_config = update_task.shared_config
+        redis = update_task.redis
+        eth_abi_values = update_task.eth_abi_values
+
+        cnode_endpoints = list(
```

Comment:
we probably don't want to make this fetch_all_registered_content_nodes request everytime because it calls the chain directly. this is expensive and slow. instead we can either
1. consume the ipfs_client.cnode_endpoints property
2. move this code to ipfs_lib.py
3. modify [this](https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/tasks/index_network_peers.py#L118) to write to a new class we create which is more generic to cache the endpoints

Reply:
yup sounds good